src/main/java/com/project/videotecha/service/impl/ProjectionServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.project.videotecha.service.impl;
+
+import com.project.videotecha.dto.ProjectionCreationDto;
+import com.project.videotecha.model.Projection;
+import com.project.videotecha.repository.ProjectionRepository;
+import com.project.videotecha.service.MovieService;
+import com.project.videotecha.service.ProjectionService;
+import com.project.videotecha.service.TheaterService;
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class ProjectionServiceImpl implements ProjectionService {
+    private final ProjectionRepository projectionRepository;
+    private final MovieService movieService;
+    private final TheaterService theaterService;
+
+    public ProjectionServiceImpl(ProjectionRepository projectionRepository, MovieService movieService,
+                                 TheaterService theaterService) {
+        this.projectionRepository = projectionRepository;
+        this.movieService = movieService;
+        this.theaterService = theaterService;
+    }
+
+    @Override
+    @Transactional
+    public Projection create(ProjectionCreationDto dto) {
+        Projection projection = new Projection(dto.getStart(), dto.getTicketPrice(),
+                movieService.getById(dto.getMovieId()), theaterService.getById(dto.getTheaterId()));
```

Comment:
Inlining another method call like this might make the code shorter, but it's harder to read. Better to fetch the movie in the separate line above and then pass it to this constructor